vertx-sql-client/src/main/java/io/vertx/sqlclient/DatabaseException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2011-2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.sqlclient;
+
+import io.vertx.core.VertxException;
+
+/**
+ * Base class for SQL Client database access failures.
```

Comment:
The doc is ambiguous and let think that it happens when we cannot access the database, instead here we want to state that this reports an exception reported by the database (hence the SQL state, etc...). _database failures_ would be fine I think.